models/Exception/DataItemNotValidException.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\ltiTestReview\models\Exception;
+
+class DataItemNotValidException extends \Exception
+{
+    public function __construct()
```

Comment:
It's generally not advised to override exception's constructors as they have sorta become part of the exception interface for use developers. You see an exception – you know that you can specify the message, code, and a parent exception.
This approach not only breaks that, but also leaves absolutely no way of specifying a parent exception, which is IMO quite important, and shouldn't be neglected, as you can easily loose a trace of it. And that, unfortunately, happens quite often in TAO codebase

What you can do instead is defined that message as a _default_ value, but still allow all the other arguments to be set or overridden.